Source/Core/Logging/FramesEventLogger.swift

```diff
@@ -14,24 +14,46 @@ protocol FramesEventLogging {
 
 final class FramesEventLogger: FramesEventLogging {
 
-    private let correlationID: String
+    private let getCorrelationID: () -> String
     private let checkoutEventLogger: CheckoutEventLogging
     private let dateProvider: DateProviding
     
     // MARK: - Init
+
+    convenience init(environment: Environment, getCorrelationID: @escaping () -> String) {
+        let checkoutEventLogger = CheckoutEventLogger(productName: Constants.productName)
+        let appBundle = Foundation.Bundle.main
+        let appPackageName = appBundle.bundleIdentifier ?? "unavailableAppPackageName"
+        let appPackageVersion = appBundle.infoDictionary?["CFBundleShortVersionString"] as? String ?? "unavailableAppPackageVersion"
```

Comment:
What would you think of having the default hardcoded values inside an enum, whether the existing `Constants` or a different one. Something like `defaultBundleIdentifier` and `defaultBundleVersion`.

Don't mind your current approach, just thinking as a practice, any hardcoded value, even a fallback, may be handy to be given sufficient visibility and grouped.